packages/starknet_provider/lib/src/model/trace_transactions_simulate.dart

```diff
@@ -0,0 +1,61 @@
+// ignore_for_file: invalid_annotation_target, non_constant_identifier_names
+
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:starknet_provider/src/model/estimate_fee.dart';
+import 'package:starknet_provider/src/model/trace_transaction_trace.dart';
+
+import 'components/block_id.dart';
+import 'fee_estimate.dart';
+import 'json_rpc_api_error.dart';
+
+part 'trace_transactions_simulate.freezed.dart';
+part 'trace_transactions_simulate.g.dart';
+
+@freezed
+class SimulateTransactions with _$SimulateTransactions {
+  const factory SimulateTransactions.result({
+    required List<SimulateTransactionResult> result,
+  }) = SimulateTransactionsResult;
+  const factory SimulateTransactions.error({
+    required JsonRpcApiError error,
+  }) = SimulateTransactionsError;
+
+  factory SimulateTransactions.fromJson(Map<String, Object?> json) =>
+      json.containsKey('error')
+          ? SimulateTransactionsError.fromJson(json)
+          : SimulateTransactionsResult.fromJson(json);
+}
+
+@JsonSerializable()
+class SimulateTransactionsRequest {
+  @JsonKey(name: 'block_id')
+  final BlockId blockId;
+  final List<BroadcastedTxn> transactions;
+  @JsonKey(name: 'simulation_flags')
+  final List<String> simulationFlags;
```

Comment:
You should use [SimulationFlag](https://github.com/focustree/starknet.dart/blob/d4da1b8d2d3ab929078e21c774031419dbba49a1/packages/starknet_provider/lib/src/model/estimate_fee.dart#L35-L48) instead